docs/01-app/03-api-reference/03-file-conventions/proxy.mdx

```diff
@@ -0,0 +1,698 @@
+---
+title: proxy.js
+description: API reference for the proxy.js file.
+related:
+  title: Learn more about Proxy
+  links:
+    - app/api-reference/functions/next-request
+    - app/api-reference/functions/next-response
+---
+
+The `proxy.js|ts` file is used to write [Proxy](/docs/app/getting-started/route-handlers-and-proxy#proxy) and run code on the server before a request is completed. Then, based on the incoming request, you can modify the response by rewriting, redirecting, modifying the request or response headers, or responding directly.
```

Comment:
I guess this link is gonna be broken, since proxy is separate from route handlers in: https://github.com/vercel/next.js/pull/84708/files